### PR TITLE
Refactor `answerGenerationNode` and test code

### PR DESCRIPTION
**Pull request by FunamaYukina**
## Why is this change needed?

I have addressed what could not be addressed in the pull request.
- https://github.com/liam-hq/liam/pull/1908

#### What was done
- ♻️ Schema Refactoring: Replaced duplicate JSON Patch operation schema definitions in `answerGenerationNode.ts` with the existing operationsSchema from `libs/schema/operationsSchema.ts `to eliminate code duplication and ensure consistency across the codebase.
- ♻️ Test Refactoring: refactored `workflow.test.ts `
- ✍️ Add Test Case: 
    - Build mode with structured JSON responses and schema changes
    - Error handling scenarios (schema update failures, network errors)
    - Edge cases (missing buildingSchemaId, malformed responses)
    - Validation of `createNewVersion` function calls with correct parameters

### 🤖 Generated by PR Agent at d334e81413a449892c9fdff11c97ce7ebcc75b03

- Refactored BuildAgent response validation to use shared `operationsSchema`.
- Enhanced chat workflow tests with helper functions and new scenarios.
  - Added tests for structured JSON responses, schema update failures, and edge cases.
  - Improved test readability and reduced redundancy.
- Updated workflow state and state management to include new properties.

## Detailed Changes

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>answerGenerationNode.ts</strong><dd><code>Use shared operationsSchema and simplify operation conversion</code></dd></summary>
<hr>

frontend/apps/app/lib/chat/workflow/nodes/answerGenerationNode.ts

<li>Replaced custom JSON Patch operation schema with shared <br><code>operationsSchema</code>.<br> <li> Simplified <code>convertToOperations</code> to directly return validated <br>operations.

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/1918/files#diff-729cc4a9c61e450edbbe4bda23b797b9ab093f901b62dd6bba3bdd063e27fa0c">+3/-26</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>langGraphWorkflow.ts</strong><dd><code>Extend ChatState with additional workflow properties</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/apps/app/lib/chat/workflow/services/langGraphWorkflow.ts

<li>Added <code>latestVersionNumber</code>, <code>organizationId</code>, and <code>userId</code> to <code>ChatState</code>.<br> <li> Updated annotations to include new state properties.

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/1918/files#diff-6c9a3d62e5f4345331b3b3278c56524650e30c1eb7d6271b698c5562a3bbe8c7">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>stateManager.ts</strong><dd><code>Update state management for new workflow properties</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/apps/app/lib/chat/workflow/services/stateManager.ts

<li>Updated state conversion functions to handle new properties.<br> <li> Changed fallback for missing <code>buildingSchemaId</code> to an empty string.

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/1918/files#diff-9f15fcf367a47cbccb7cd9e0742e2e3fde55cd5b0d8d921f9c5e4b36a509794a">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>workflow.test.ts</strong><dd><code>Refactor and expand chat workflow test coverage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

frontend/apps/app/lib/chat/workflow/workflow.test.ts

<li>Introduced helper functions for mock schema and base state creation.<br> <li> Added comprehensive tests for Build mode, error handling, and state <br>management.<br> <li> Mocked <code>createNewVersion</code> and validated its usage in tests.<br> <li> Reduced redundancy and improved test clarity and coverage.

</details>

  </td>
  <td><a href="https://github.com/liam-hq/liam/pull/1918/files#diff-36093fb825c1f79037cf8255a6bf3ed8cc8ddb9d5d5f342dba067701d7768e2b">+339/-230</a></td>

</tr>
</table></td></tr></tr></tbody></table>

## Additional Notes

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: frontend/apps/app/lib/chat/workflow/nodes/answerGenerationNode.ts [hunk 1]
```diff
@@ -72,23 +65,7 @@ const convertToOperations = (
...
-      ...baseOperation,
-      ...(change.value !== undefined && { value: change.value }),
-      ...(change.from !== undefined && { from: change.from }),
-    }
-
-    // Type assertion is safe here because:
-    // 1. valibot has already validated the structure
-    // 2. We're following JSON Patch RFC 6902 specification
-    return operation as Operation
+    return change
```
Comment (MH4GF) [inline frontend/apps/app/lib/chat/workflow/nodes/answerGenerationNode.ts:68]: If you no longer do anything, then you should not need this function.

**Review by MH4GF — APPROVED** [review]
LGTM 👍🏻

**Review by hoshinotsuyoshi — COMMENTED** [review]
thx!
